distributed/comm/tcp.py

```diff
@@ -371,6 +354,79 @@ def extra_info(self):
         return self._extra
 
 
+async def read_bytes_rw(stream: IOStream, n: int) -> memoryview:
+    """Read n bytes from stream. Unlike stream.read_bytes, allow for
+    very large messages and return a writeable buffer.
+    """
+    buf = host_array(n)
+
+    for i, j in sliding_window(
+        2,
+        range(0, n + OPENSSL_MAX_CHUNKSIZE, OPENSSL_MAX_CHUNKSIZE),
+    ):
+        chunk = buf[i:j]
+        chunk_nbytes = chunk.nbytes
+        n = await stream.read_into(chunk)  # type: ignore[arg-type]
+        assert n == chunk_nbytes, (n, chunk_nbytes)
```

Comment:
Apologies for any naive things here, for my own understanding, this assert is a sanity check, as [according to the docs for read_into](https://www.tornadoweb.org/en/stable/iostream.html#tornado.iostream.BaseIOStream.read_into) this won't return until `chunk` (of which is `chunk_nbytes` in length) is completely filled. Is there a potential that it won't be filled due to chunk being larger than any remaining bytes, and thus sit idle?

I would suppose not judging by the caller setting `n` but got slightly confused with the openssl sizes and that may make a chunk larger than any remaining stream data.

Reply:
Or now that I look at it again.. bytes in the stream would always be larger than the buffer, and thus stream would always have bytes to write given we're accounting for OpenSSL sizing. Do I have that right?

Reply:
yes that assertion is just defensive programming.